src/Writers/CSharpWriter/ConcreteNavigationAccessorProperty.cs

```diff
@@ -12,8 +12,8 @@ public class ConcreteNavigationAccessorProperty : NavigationProperty
         protected ConcreteNavigationAccessorProperty(OdcmProperty odcmProperty) : base(odcmProperty)
         {
             FieldName = NamesService.GetPropertyFieldName(odcmProperty);
-            InstanceType = NamesService.GetFetcherTypeName(odcmProperty.Type);
-            Type = new Type(NamesService.GetConcreteTypeName(odcmProperty.Type));
+            InstanceType = NamesService.GetFetcherTypeName(odcmProperty.Projection.Type);
```

Comment:
It would be great to have this temporary scafolding handled by an extension method on odcmProperty. It would be bad if folks took this pattern as an intended long-term practice.